### PR TITLE
Introduce a function to compute the doubling/halving time for a lightcurve

### PR DESCRIPTION
**Pull request by cgalelli**
This pull request is a simple reopening of pull request #4727 after it was closed by mistake. It contains the same implementation as the previous PR: the flux doubling computation for simple numpy arrays and a wrapper for lightcurve objects, and associated tests.

### CODE REVIEW

File: gammapy/estimators/utils.py [hunk 1]
```diff
@@ -386,3 +387,79 @@ def compute_lightcurve_fpp(lightcurve, flux_quantity="flux"):
...
+    rapid variations are clearly evident: for example it is useful in AGN flaring episodes.
+
+    This quantity, especially for AGN flares, is often expressed
+    as the pair of doubling time and halving time, or the minimum characteristic time
+    for the rising and falling components respectively.
+
+    Parameters
+    ----------
+    lightcurve : `~gammapy.estimators.FluxPoints`
+        the lightcurve object
```
Comment (bkhelifi) [inline gammapy/estimators/utils.py:409]:
```suggestion
        The lightcurve object.
```

File: gammapy/estimators/utils.py [hunk 2]
```diff
@@ -386,3 +387,79 @@ def compute_lightcurve_fpp(lightcurve, flux_quantity="flux"):
...
+    axis_name : str
+        name of the axis over which to compute the flux doubling
```
Comment (bkhelifi) [inline gammapy/estimators/utils.py:411]:
```suggestion
        Name of the axis over which to compute the flux doubling.
```

File: gammapy/estimators/utils.py [hunk 3]
```diff
@@ -386,3 +387,79 @@ def compute_lightcurve_fpp(lightcurve, flux_quantity="flux"):
...
+    flux_quantity : str
+        flux quantity to use for calculation. Should be 'dnde', 'flux', 'e2dnde' or 'eflux'. Default is 'flux'.
+        Useful in case of custom lightcurves computed outside gammapy
```
Comment (bkhelifi) [inline gammapy/estimators/utils.py:414]:
```suggestion
        Flux quantity to use for calculation. Should be 'dnde', 'flux', 'e2dnde' or 'eflux'.
        Default is 'flux'.
```

File: gammapy/stats/variability.py [hunk 4]
```diff
@@ -137,3 +138,84 @@ def compute_chisq(flux):
...
+     .. math::
+        doubling = min(\frac{t_(i+1)-t_i}{log_2{f_(i+1)/f_i}})
+
+    Where f_i and f_(i+1) are the fluxes measured at subsequent coordinates t_i and t_(i+1)
+    The error is obtained by propagating the relative errors on the flux measures.
+
+    Parameters
+    ----------
+    flux : `~astropy.units.Quantity`
+        the measured fluxes
```
Comment (bkhelifi) [inline gammapy/stats/variability.py:161]:
```suggestion
        The measured fluxes.
```

File: gammapy/stats/variability.py [hunk 5]
```diff
@@ -137,3 +138,84 @@ def compute_chisq(flux):
...
+
+    Computing the flux doubling can give the doubling time in a lightcurve
+    displaying significant temporal variability, e.g. an AGN flare.
+
+    The variable is computed as:
+
+     .. math::
+        doubling = min(\frac{t_(i+1)-t_i}{log_2{f_(i+1)/f_i}})
+
+    Where f_i and f_(i+1) are the fluxes measured at subsequent coordinates t_i and t_(i+1)
```
Comment (bkhelifi) [inline gammapy/stats/variability.py:155]:
```suggestion
    where f_i and f_(i+1) are the fluxes measured at subsequent coordinates t_i and t_(i+1).
```

File: gammapy/stats/variability.py [hunk 6]
```diff
@@ -137,3 +138,84 @@ def compute_chisq(flux):
...
+    The error is obtained by propagating the relative errors on the flux measures.
+
+    Parameters
+    ----------
+    flux : `~astropy.units.Quantity`
+        the measured fluxes
+    flux_err : `~astropy.units.Quantity`
+        the error on measured fluxes
+    coords: `~astropy.units.Quantity`
+        the coordinates at which the fluxes are measured
```
Comment (bkhelifi) [inline gammapy/stats/variability.py:165]:
```suggestion
        The coordinates at which the fluxes are measured.
```

File: gammapy/stats/variability.py [hunk 7]
```diff
@@ -137,3 +138,84 @@ def compute_chisq(flux):
...
+
+    Where f_i and f_(i+1) are the fluxes measured at subsequent coordinates t_i and t_(i+1)
+    The error is obtained by propagating the relative errors on the flux measures.
+
+    Parameters
+    ----------
+    flux : `~astropy.units.Quantity`
+        the measured fluxes
+    flux_err : `~astropy.units.Quantity`
+        the error on measured fluxes
```
Comment (bkhelifi) [inline gammapy/stats/variability.py:163]:
```suggestion
        The error on measured fluxes.
```

File: gammapy/stats/variability.py [hunk 8]
```diff
@@ -137,3 +138,84 @@ def compute_chisq(flux):
...
+    coords: `~astropy.units.Quantity`
+        the coordinates at which the fluxes are measured
+    axis : int, optional
+        Axis along which the value is computed.
+
+    Returns
+    -------
+    doubling, doubling_err, coord : `~numpy.ndarray`
```
Comment (bkhelifi) [inline gammapy/stats/variability.py:171]: In fact, the function returns 3 ndarrays, not one.
The first ndarray contains doubling and halving flux, the second their corresponding errors, and the 2 coordinate values at which it happens

**Review by bkhelifi — COMMENTED** [review]
Here are some comments... We will continue tomorrow.

**Review by registerrier — APPROVED** [review]
Thanks @cgalelli . 

I think the implementation could still be simplified. But we can keep this for later. Maybe add a #TODO

**Pull request merged**